src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -338,22 +339,31 @@ object DeBruijnGraph {
   type Sequence = DeBruijnGraph#Sequence
   type Kmer = DeBruijnGraph#Kmer
 
-  def apply(sequences: Seq[Sequence],
+  def apply(reads: Seq[Read],
             kmerSize: Int,
             minOccurrence: Int = 1,
+            minMeanKmerBaseQuality: Int = 0,
             mergeNodes: Boolean = false): DeBruijnGraph = {
 
     val kmerCounts = mutable.Map.empty[DeBruijnGraph#Kmer, Int]
 
-    sequences
-      .filter(Bases.allStandardBases(_))
-      .foreach(
-        _.sliding(kmerSize)
-          .foreach(seq => {
-            val count = kmerCounts.getOrElse(seq, 0)
-            kmerCounts.update(seq, count + 1)
-          })
-      )
+    reads.foreach(
+      read => {
+        read.sequence
+          .sliding(kmerSize)
+          .zipWithIndex
+          .foreach(
+            { case (seq, index) =>
+              val baseQualities = read.baseQualities.slice(index, index + kmerSize).map(_.toFloat)
+              val meanBaseQuality = mean(baseQualities)
```

Comment:
follow-up question: is it worth optimizing this to compute/maintain a "streaming" mean, so that you aren't averaging (k-1) of the same qualities at each position that you just did at the previous position?